client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/SecurityDocumentationIT.java

```diff
@@ -496,4 +499,77 @@ private void addRole(String roleName) throws IOException {
         client().performRequest(addRoleRequest);
     }
 
+    public void testCreateToken() throws Exception {
+        RestHighLevelClient client = highLevelClient();
+
+        {
+            // Setup user
+            PutUserRequest putUserRequest = new PutUserRequest("token_user", "password".toCharArray(),
+                Collections.singletonList("kibana_user"), null, null, true, null, RefreshPolicy.IMMEDIATE);
+            PutUserResponse putUserResponse = client.security().putUser(putUserRequest, RequestOptions.DEFAULT);
+            assertTrue(putUserResponse.isCreated());
+        }
+        {
+            // tag::create-token-password-request
+            final char[] password = new char[]{'p', 'a', 's', 's', 'w', 'o', 'r', 'd'};
+            CreateTokenRequest createTokenRequest = CreateTokenRequest.passwordGrant("token_user", password);
+            // end::create-token-password-request
+
+            // tag::create-token-execute
+            CreateTokenResponse createTokenResponse = client.security().createToken(createTokenRequest, RequestOptions.DEFAULT);
+            // end::create-token-execute
+
+            // tag::create-token-response
+            String accessToken = createTokenResponse.getAccessToken();    // <1>
+            String refreshToken = createTokenResponse.getRefreshToken();    // <2>
+            // end::create-token-response
+            assertNotNull(accessToken);
+            assertNotNull(refreshToken);
+            assertNotNull(createTokenResponse.getExpiresIn());
+
+            // tag::create-token-refresh-request
+            createTokenRequest = CreateTokenRequest.refreshTokenGrant(refreshToken);
+            // end::create-token-refresh-request
+
+            CreateTokenResponse refreshResponse = client.security().createToken(createTokenRequest, RequestOptions.DEFAULT);
+            assertNotNull(refreshResponse.getAccessToken());
+            assertNotNull(refreshResponse.getRefreshToken());
+        }
+
+        {
+            // tag::create-token-client-credentials-request
+            CreateTokenRequest createTokenRequest = CreateTokenRequest.clientCredentialsGrant();
+            // end::create-token-client-credentials-request
+
+            ActionListener<CreateTokenResponse> listener;
+            //tag::create-token-execute-listener
+            listener = new ActionListener<CreateTokenResponse>() {
+                @Override
+                public void onResponse(CreateTokenResponse createTokenResponse) {
+                    // <1>
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    // <2>
+                }
+            };
+            //end::create-token-execute-listener
+
+            // Replace the empty listener by a blocking listener in test
+            final CountDownLatch latch = new CountDownLatch(1);
```

Comment:
why use both a latched action listener and a plain action future?

Reply:
Because my brain wasn't working very well.